### PR TITLE
Added test to cover the installation of aliased plugins

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

[rn:skip]

## What does this PR do?

Adds test coverage for plugin manager when it installs aliased plugins

## Why is it important/What is the impact to the user?

No impact for the user

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: spec/unit/plugin_manager/install_spec.rb [hunk 1]
```diff
@@ -40,6 +40,56 @@
...
+      expect(cmd).to receive(:remove_unused_locally_installed_gems!).and_return(nil)
+      cmd.verify = true
+    end
+
+    it "should not consider as valid plugin a gem with an alias name" do
+#       expect(cmd).to receive(:validate_cli_options!).and_return(nil)
+#       expect(cmd).to receive(:plugins_gems).and_return([["logstash-input-elastic_agent", nil]])
+#       expect(cmd).to receive(:install_gems_list!).and_return(nil)
+#       expect(cmd).to receive(:remove_unused_locally_installed_gems!).and_return(nil)
+#       cmd.verify = true
```
[the remark this answers is not in the document]
Comment (andsel) [reply spec/unit/plugin_manager/install_spec.rb]: Thank you @jsvd, my bad, missed to polish the commented stuff

File: spec/unit/plugin_manager/install_spec.rb [hunk 2]
```diff
@@ -40,6 +40,56 @@
     end
   end
 
+  context "when installs alias plugin" do
+
+    before(:each) do
+      expect(cmd).to receive(:validate_cli_options!).and_return(nil)
+      expect(cmd).to receive(:plugins_gems).and_return([["logstash-input-elastic_agent", nil]])
+      expect(cmd).to receive(:install_gems_list!).and_return(nil)
+      expect(cmd).to receive(:remove_unused_locally_installed_gems!).and_return(nil)
```
[the remark this answers is not in the document]
Comment (andsel) [reply spec/unit/plugin_manager/install_spec.rb]: are mainly inherited by the above test that verify the installation of a dummy gem, but in this case uses a real alias name shipped into the `plugin_aliases.yml` file

[the remark this answers is not in the document]
Comment (andsel) [reply spec/unit/plugin_manager/install_spec.rb]: mainly those mocks are used to provide data to command (in an indirect form) and to control the path of execution go straight to the `verify_remote!` method

**Review by jsvd — APPROVED** [review]
LGTM

**Pull request merged**